packages/playground/src/weblets/tf_domains.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <weblet-layout ref="layout" title-image="images/icons/domains.png">
+    <template #title>Deploy a Domains Instance </template>
+
+    <d-tabs :tabs="[{ title: 'Config', value: 'config' }]">
+      <input-tooltip tooltip="Selecting custom domain sets subdomain as gateway name.">
+        <input-validator
+          :value="subdomain"
+          :rules="[
+            validators.required('Subdomain is required.'),
+            validators.isLowercase('Subdomain should consist of lowercase letters only.'),
+            validators.isAlphanumeric('Subdomain should consist of letters and numbers only.'),
+            subdomain => validators.isAlpha('Subdomain must start with alphabet char.')(subdomain[0]),
+            validators.minLength('Subdomain must be at least 4 characters.', 4),
+            subdomain => validators.maxLength('Subdomain cannot exceed 50 characters.', 50)(subdomain),
```

Comment:
can we add async rule to check the subdomain availability like here? https://github.com/threefoldtech/tfgrid-sdk-ts/blob/03ea91378fa3b74ceb5a60a0c6228e14f9559196/packages/playground/src/components/manage_gateway_dialog.vue#L96